### PR TITLE
DEV-120: fix bugs in search 

### PR DESCRIPTION
**Pull request by mpark63**
Backend PR: https://github.com/uCredit-Dev/uCredit-API/pull/77

## Context 
Some PRs were merged into dev such as #413 #430 in frontend and https://github.com/uCredit-Dev/uCredit-API/pull/70 https://github.com/uCredit-Dev/uCredit-API/pull/72 and https://github.com/uCredit-Dev/uCredit-API/pull/64 in backend. However, we are noticing some bugs in development branch which we need to address before merging these changes into master. (need to do this asap because development branch is lagging behind a lot! aka the JWT changes still didn't make it to production). 

DEV-120 task was originally to do with improving the speed of the new search route, more on that in https://github.com/uCredit-Dev/uCredit-API/pull/77. 

## Updates 
- Degree progress calculation crashes client 
  - handle course.areas undefined case
- Migrated cart search to backend (now one API call) 
  - call new route from frontend and handle pagination from response 
- Ethics of AI was not counting for Computer ethics requirement 
  - fix typo in majors.tsx
- colors were wrong for courses 
  - rename areas to be area in dev database 

## Testing 
Localhost with backend server on branch [DEV-120](https://github.com/uCredit-Dev/uCredit-API/pull/77)

## todos 
address my own comments (remove console.log, run prettier, correct areas to area etc...)

### CODE REVIEW

File: lib/components/dashboard/degree-info/distributionFunctions.tsx [hunk 1]
```diff
@@ -101,7 +101,9 @@ const handleTagType = (
       break;
     case 'A': // Area
       updatedConcat = (
-        course.areas !== 'None' && course.areas.includes(splitArr[index])
+        course.areas !== undefined &&
+        course.areas !== 'None' &&
+        course.areas.includes(splitArr[index])
```
Comment (mpark63) [inline lib/components/dashboard/degree-info/distributionFunctions.tsx:106]: am now thinking this should be area not areas. It's area in main/dev but areas in DIS branch, hence the confusion.

File: lib/components/popups/course-search/cart/CartCourseList.tsx [hunk 2]
```diff
@@ -1,62 +1,42 @@
...
-    });
-    // If coursesPerPage doesn't divide perfectly into total courses, we need one more page.
-    const division = Math.floor(filteredCourses.length / coursesPerPage);
-    const pages =
-      filteredCourses.length % coursesPerPage === 0 ? division : division + 1;
-    setPageCount(pages);
-    setFilteredCourses(filteredCourses);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [courses, props.textFilter]);
+    console.log(courses);
```
Comment (mpark63) [inline lib/components/popups/course-search/cart/CartCourseList.tsx:38]: remove console.log

File: lib/resources/majors.tsx [hunk 3]
```diff
@@ -3404,7 +3414,7 @@ const bsCS_New: Major = {
             'EN.601.124 The Ethics of Artificial Intelligence and Automation (The Ethics of Artificial Intelligence and Automation) <br /> ' +
             'EN.660.400 Practical Ethics for Future Leaders',
           required_credits: 1,
-          criteria: 'EN.600.104[C]^OR^EN.601.104[C]^OR^EN.660.400[C]',
+          criteria: 'EN.600.104[C]^OR^EN.601.124[C]^OR^EN.660.400[C]',
```
Comment (mpark63) [inline lib/resources/majors.tsx:3417]: was a bug

**Review by mattliu-mygit — APPROVED** [review]
Awesome, amazing work ty!

File: lib/components/dashboard/degree-info/distributionFunctions.tsx (hunk 1 above)
Comment (mpark63) [reply lib/components/dashboard/degree-info/distributionFunctions.tsx:106]: nm

**Pull request merged**